### ISSUE TITLE
Error on parsing output (It might non-JSON output) : ""

### ISSUE THREAD

**Issue by tophattom** [misogi/vscode-ruby-rubocop#22]
After the latest update I get this error every time rubocop is run.

`Warn: Error on parsing output (It might non-JSON output) : ""`

**Comment by tophattom** [comment]
This was an issue with my configuration

**Comment by robertocortiz** [comment]
I'm curious what configuration setting did you have to fix to get this to go away?

**Comment by tophattom** [comment]
I had an issue with rubocop not being found due to using different Ruby version with rvm. Sorry, but I cannot quite remember what did to fix this exactly.

**Comment by cesmoak** [comment]
I got rid of this error as soon as I specified a rubocop config file in the extension settings (`"ruby.rubocop.configFilePath": "<something>"`) and ensured that that file actually existed (an empty file is fine).

**Comment by wulftone** [comment]
Not working for me, tried all the above solutions, installed it on every RVM Ruby, installed on system Ruby... it worked fine up until a week or two ago (don't recall what changed, unfortunately).  How can I debug the command being run and the output to ensure it is doing the right thing?

**Comment by alfonsoperez** [comment]
Yeah, this is happening to me too, already specified a config file

**Comment by mjbrender** [comment]
Same issue for me. Using RVM to manage versions, but installed rubocop all over the place. If you don't mind @tophattom, I might ask to open this back up given the now-4 of us still figuring this out. 

Or @misogi if you prefer a new issue I'm happy to open one. Thanks!

**Comment by misogi** [comment]
@mjbrender Still I have not been identify this issue.

It may only RVM(path management) causes this issue or not.

**Comment by misogi** [comment]
This is my environment. How about RVM?

```
❯ which rubocop
/Users/misogi/.rbenv/shims/rubocop
```

**Comment by mjbrender** [comment]
Hey @misogi - here is mine: 

```
> which rubocop
/Users/mjbrende/.rvm/gems/ruby-2.2.5/bin/rubocop
```

**Comment by noiseunion** [comment]
I am having the same issue with no luck.  My config looks like this:

```json
{
  "ruby.rubocop.executePath": "/Users/ninjago/.rvm/gems/ruby-2.3.1@raceday/bin/",
  "ruby.rubocop.configFilePath": "/Users/ninjago/code/raceday/.rubocop.yml",
  "ruby.rubocop.onSave": true
}
```

I am using RVM and Rubocop is installed in the projects gemset.  I can run Rubocop just fine from the command line.

**Comment by noiseunion** [comment]
if this is helpful as well.

```bash
> which rubocop
/Users/ninjago/.rvm/gems/ruby-2.3.1@raceday/bin/rubocop
```

So just as a note, my rubocop is installed in a gemset (raceday) for ruby 2.3.1

**Comment by magegu** [comment]
I started to see this issue after I added the following to my .rubocop.yml project file.

```
AllCops:
  TargetRubyVersion: 2.3.1
```

**Comment by paulewetzel** [comment]
If I remove "ruby.rubocop.configFilePath", the error goes away.

**Comment by paulewetzel** [comment]
I was looking at this code below to see if my args were going in correctly. I abbreviated the path , but this is what it returns and the paths are correct.
 **args:** /home/user/app/package.rb,--format,json,--config,/home/user/app/.rubocop

> let args = this.commandArguments(fileName);
>  return cp.execFile(executeFile, args, { cwd: currentPath }, onDidExec);

Just wanted to confirm that the args looked OK.

**Comment by paulewetzel** [comment]
UPDATE: ran rubocop on the command line, noticed it was looking for enabled and disabled config files and erring out, so I added them. Then I saw there were some cops that it did not like, so I took them out of my config files. Now the error message has changed, and the content is clearly not JSON. 

Error on parsing output (It might non-JSON output) : ">>>>>>> Style/AccessModifierIndentation <<<<<<<<<< >>>>>>> Style/AccessorMethodName <<<<<<<<<< >>>>>>> Style/Alias <<<<<<<<<< >>>>>>

I was able to see the cops it did not like by debug code to rubo cop where it was iterating through the cops. Not sure if this helps.

**Comment by paulewetzel** [comment]
Progress: Getting actual JSON back now from line 40 of /out/rubocop.js
let output = stdout.toString();
            vscode.window.showWarningMessage(output);
{"metadata":{"rubocop_version":"0.46.0","ruby_engine":"ruby","ruby_version":"2.2.3","ruby_patchlevel":"173","ruby_platform":"x86_64-linux"},"files":[{"path":"app/service/fucked_json_service.rb","offense

**Comment by paulewetzel** [comment]
Now It works using the task command, formats my code correctly!!! F*** Yeah!!

**Comment by elgreco84** [comment]
I switched from rvm to rbenv to get it working

**Comment by rhys-vdw** [comment]
> TargetRubyVersion: 2.3.1

@magegu FYI This is actually an invalid configuration option.

See bbatsov/rubocop#2663 for more information. They've marked it as a wontfix. I'd encourage you to share your thoughts in that issue thread.

**Comment by rhys-vdw** [comment]
Also can't get rid of this issue. I get the message every time I save, but no rubocop linting is performed.

```
    "ruby.rubocop.executePath": "/Users/rhys/.gem/ruby/2.3.3/bin/",
    "ruby.rubocop.configFilePath": "/Users/rhys/Projects/usability-hub/usability_hub/.rubocop.yml",
    "ruby.rubocop.onSave": true
```

I'm using [chruby](https://github.com/postmodern/chruby).

**Comment by citystar** [comment]
I ran `rubocop -d` in console to figure out that empty config file is invalid. Putting a valid config file to ~/.rubocop.yml solved my problem.

**Comment by rhys-vdw** [comment]
I ended up just not using this package and using [`vscode-ruby`](https://github.com/rubyide/vscode-ruby) instead. All it needed was a `rubocop: true` option and it just works (also continuously checks instead of requiring save). I'm not sure what other features this package offers but I couldn't get it to do anything at all (except display that error message).

**Comment by misogi** [comment]
Sorry. I have no idea for this problem.
I can't identify what causes this error. (on rvm or chruby or invalid .rubocop.yml)
You should use vscode-ruby instead.

I made an update for output stderr but no solution.

**Comment by eshadd** [comment]
I had this error with some code I inherited. It seems that there were just flat out too many "violations". Once I did autocorrect, everything seems to work as expected with no JSON error. Only problem is that autocorrect seems to autosave as well (i.e. it overwrote my pre-autocorrect version).

Maybe the JSON has a limit?

**Comment by mohgabr** [comment]
like @eshadd said it is because it has a limit just try to autocorrect from terminal

**Comment by Jonic** [comment]
I've just encountered this error. The issue in my case was a `RUBOCOP_OPTS` env var in my terminal profile which was overriding the default formatter. This var is a bit of a hidden feature of Rubocop - it's only mentioned in passing on this page: https://docs.rubocop.org/rubocop/usage/basic_usage.html#command-line-flags (search for `RUBOCOP_OPTS`).

Setting this env var can potentially cause conflicts with the way Rubocop is executed via this extension. It looks to me that `Rubocop.execute` would have to scan for and discard any formatting options that were passed to it (https://github.com/misogi/vscode-ruby-rubocop/blob/master/src/rubocop.ts#L177-L180), or override the `RUBOCOP_OPTS` env var before execution.

If this is also effecting you, you can get around it by adding the format options to a `.rubocop` file in your project instead. Note the lack of `.yml` extension there - this file is also mentioned in the same docs as `RUBOCOP_OPTS`.
